logging/logging-annotations/src/main/java/org/infinispan/logging/annotations/Description.java

```diff
@@ -0,0 +1,35 @@
+package org.infinispan.logging.annotations;
+
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.CLASS;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * Descriptions should provide additional detail for each message and suggested user action.
+ * 
+ * Description.
+ * @author Durgesh Anaokar
+ * @since 13.0
+ */
+
+@Retention(CLASS)
+@Target(METHOD)
+public @interface Description {
+
+   /**
+    * The default format string of this message.
+    * <p>
+    * Expressions in the form of {@code ${property.key:default-value}} can be used for the value. If the property key is
+    * prefixed with {@code sys.} a {@linkplain System#getProperty(String) system property} will be used. If the key is
+    * prefixed with {@code env.} an {@linkplain System#getenv(String) environment variable} will be used. In all other cases
+    * the {@code org.jboss.logging.tools.expressionProperties} processor argument is used to specify the path the properties
+    * file which contains the values for the expressions.
+    * </p>
+    *
+    * @return the format string
+    */
```

Comment:
I'm not sure these docs are correct. In the code it seems like we just use `.value()` as a simple `String` and do not resolve properties or env variables. We should update the Javadoc to reflect this. We can always add support for properties later if required.

Reply:
This still needs addressing.